### ISSUE TITLE
oidc time format issues / aws-sdk2-java

### ISSUE THREAD

**Issue by thisismana** [99designs/aws-vault#655]
Using the `--server` feature the AWS SDK fails when trying to parse the Expiration format used for oidc, the same date format is used for `aws-vault exec $PROFILE --`

`java.time.format.DateTimeParseException: Text '2020-09-10T18:16:52+02:00' could not be parsed at index 19` (char 19 refers to the `+` symbol).

```sh
$ aws-vault exec $OIDC_PROFILE -- env | grep AWS_SESSION_EXP
AWS_SESSION_EXPIRATION=2020-09-10T18:16:52+02:00
```

when doing so using classic PROFILES (ACCESS_KEY, SECRET_ACCESS_KEY), the correct format is used:
```sh
aws-vault exec $CLASSIC_PROFILE -- env | grep AWS_SESSION_EXP
AWS_SESSION_EXPIRATION=2020-09-10T10:45:32Z
```
In the code `RFC3339` explicitly used, but it seems `ISO-8601` with timezone `UTC`is expected.

https://github.com/99designs/aws-vault/blob/2e8df9e12680c2f124f9aa578da0e16ac121a358/vault/oidctokenkeyring.go#L87

I've never coded in go, but if this seems to be legitimate issue, I'd be glad to try it myself.

- [x] I am using the latest release of AWS Vault
- [ ] I have provided my `.aws/config` (redacted if necessary)
- [x] I have provided the debug output using `aws-vault --debug` (redacted if necessary)

debug output:
<details>
<pre>
2020/09/10 11:47:14 aws-vault v6.0.0
2020/09/10 11:47:14 Loading config file /Users/xxx/.aws/config
2020/09/10 11:47:14 Parsing config file /Users/xxx/.aws/config
2020/09/10 11:47:14 [keyring] Considering backends: [keychain]
2020/09/10 11:47:14 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/09/10 11:47:14 [keyring] Found 1 results
2020/09/10 11:47:14 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/09/10 11:47:14 [keyring] Found 1 results
2020/09/10 11:47:14 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/09/10 11:47:14 [keyring] Found 1 results
2020/09/10 11:47:14 [keyring] Querying keychain for service="aws-vault", account="sso.GetRoleCredentials,foo,bar,-62135596800", keychain="aws-vault.keychain"
2020/09/10 11:47:14 [keyring] No results found
2020/09/10 11:47:14 [keyring] Querying keychain for service="aws-vault", account="oidc:https://d-xxx.awsapps.com/start", keychain="aws-vault.keychain"
2020/09/10 11:47:14 [keyring] No results found
2020/09/10 11:47:15 Created new OIDC client (expires at: 2020-12-09 10:47:15 +0100 CET)
2020/09/10 11:47:15 Created OIDC device code for https://d-xxx.awsapps.com/start (expires in: 600s)
2020/09/10 11:47:15 Opening SSO authorization page in browser
Opening the SSO authorization page in your default browser (use Ctrl-C to abort)
https://device.sso.eu-west-1.amazonaws.com/?user_code=xxx
2020/09/10 11:47:24 Created new OIDC access token for https://d-xxx.awsapps.com/start (expires in: 28800s)
2020/09/10 11:47:24 [keyring] Checking keychain status
2020/09/10 11:47:24 [keyring] Keychain status returned nil, keychain exists
2020/09/10 11:47:24 [keyring] Keychain item trusts keyring
2020/09/10 11:47:24 [keyring] Adding service="aws-vault", label="aws-vault oidc token for https://d-xxx.awsapps.com/start (expires 2020-09-10T19:47:24+02:00)", account="oidc:https://d-xxx.awsapps.com/start", trusted=true to osx keychain "aws-vault.keychain"
2020/09/10 11:47:24 Got credentials ****************K5DR for SSO role ROLE (account: xxx), expires in 7h59m58.160075s
2020/09/10 11:47:24 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/09/10 11:47:24 [keyring] Found 2 results
2020/09/10 11:47:24 [keyring] Querying keychain for service="aws-vault", keychain="aws-vault.keychain"
2020/09/10 11:47:24 [keyring] Found 2 results
2020/09/10 11:47:24 [keyring] Checking keychain status
2020/09/10 11:47:24 [keyring] Keychain status returned nil, keychain exists
2020/09/10 11:47:24 [keyring] Keychain item trusts keyring
2020/09/10 11:47:24 [keyring] Adding service="aws-vault", label="aws-vault session for to (expires 2020-09-10T19:47:23+02:00)", account="sso.GetRoleCredentials,foo,bar,xxx" trusted=true to osx keychain "aws-vault.keychain"
</pre>
</details>

**Comment by mtibben** [comment]
Hmm this switch to RFC3339 happened in https://github.com/99designs/aws-vault/pull/600, @ajkerrigan any thoughts?

My understanding was that RFC3339 was compatible but a more strict version of the older ISO8601, but sounds like that might not be the case?

**Comment by ajkerrigan** [comment]
I took a look at this, and I think I see what's going on. The change in #600 was about using consistent formatting during _output_. But it looks like SSO token/credential expiration times are stored in local time while session cred expirations are stored as UTC.

I've opened #656 to address this by storing SSO expiration times in UTC. Let me know what you think, and if there's a better way to handle it.

**More Detail**

This seems partly due to differences in the underlying SDK, where sts [Credentials](https://docs.aws.amazon.com/sdk-for-go/api/service/sts/#Credentials) returns its Expiration as `*time.Time` (in UTC) while sso [RoleCredentials](https://docs.aws.amazon.com/sdk-for-go/api/service/sso/#RoleCredentials) gives `*int64`. When storing SSO credentials, we're using either `time.Now()` or `aws.MillisecondsTimeValue`. Both give us a local time.
